core/c4diagram.go

```diff
@@ -0,0 +1,355 @@
+// package defines diagram as code using the graph input
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/kislerdm/diagramastext/core/compression"
+)
+
+type optionsPlantUMLClient struct {
+	httpClient HttpClient
+}
+
+type clientPlantUML struct {
+	options optionsPlantUMLClient
+
+	baseURL string
+}
+
+func (c *clientPlantUML) Do(v DiagramGraph) (ResponseDiagram, error) {
```

Comment:
**suggestion:** we should support `context.Context` in this function, so the caller has the opportunity to control the execution flow of the whole function instead of just having a HTTP timeout control.

Reply:
Great suggestion! 👍🏻  Let's convert it to an issue, and implement post initial launch.

Reply:
Implemented